src/query/builder.ts

```diff
@@ -259,7 +259,16 @@ export class QueryBuilder<
       this.project,
       this.restricted,
       [...this.filters, filter]
-    )
+    ) as unknown as Omit<
+      QueryBuilder<
+        Schema,
+        Mappings,
+        Type,
+        Project,
+        Exclude | 'filter'
+      >,
+      Exclude | 'filter'
```

Comment:
I don't think we need to exclude filters - we should be able to chain `.filter()` calls.

Reply:
Makes sense, updated